tests/core/data/test_aws_s3_data_node.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+from dataclasses import dataclass
+from unittest.mock import patch
+
+import boto3
+from moto import mock_s3
+import pytest
+
+from src.taipy.core.data.data_node_id import DataNodeId
+from src.taipy.core.data.aws_s3 import S3ObjectDataNode
+from src.taipy.core.exceptions.exceptions import InvalidCustomDocument, MissingRequiredProperty
+from taipy.config.common.scope import Scope
+
+
+
+class TestS3ObjectDataNode:
+    __properties = [
+        {
+            "aws_access_key": "testing",
+            "aws_secret_acces_key": "testing",
+            "aws_region": "us-east-1",
+            "aws_s3_bucket_name":"taipy",
+            "aws_s3_object_key":"taipy-object",
+            "db_extra_args": {},
+        }
+    ]
+
+    @mock_s3
+    @pytest.mark.parametrize("properties", __properties)
+    def test_create(self, properties):
+        aws_s3_object_dn = S3ObjectDataNode(
+            "foo_bar_aws_s3",
+            Scope.SCENARIO,
+            properties=properties,
+        )
+        assert isinstance(aws_s3_object_dn, S3ObjectDataNode)
+        assert aws_s3_object_dn.storage_type() == "aws_s3"
+        assert aws_s3_object_dn.config_id == "foo_bar_aws_s3"
+        assert aws_s3_object_dn.scope == Scope.SCENARIO
+        assert aws_s3_object_dn.id is not None
+        assert aws_s3_object_dn.owner_id is None
+        assert aws_s3_object_dn.job_ids == []
+        assert aws_s3_object_dn.is_ready_for_reading
+
+
+    @mock_s3
+    @pytest.mark.parametrize('data', [('Hello, world!'),])
+    @pytest.mark.parametrize("properties", __properties)
+    def test_write(self, properties, data ):
+        bucket_name = properties["aws_s3_bucket_name"]
+
+        # Create an S3 client
+        s3_client = boto3.client('s3')
+
+        # Create a bucket
+        s3_client.create_bucket(Bucket=bucket_name)
+
+        aws_s3_object_dn = S3ObjectDataNode("foo_aws_s3", Scope.SCENARIO, properties=properties)
+        aws_s3_object_dn._write(data)
+
+        response = aws_s3_object_dn._read()
+
+        assert response == data
+
+
+    @mock_s3
+    @pytest.mark.parametrize('data', [('Hello, other world!'),])
+    @pytest.mark.parametrize("properties", __properties)
+    def test_read(self, properties, data ):
```

Comment:
This test is no different from the `test_write` method.

The test should do the following:
1. Connect to the bucket and write object to the bucket without using Taipy.
2. Create S3ObjectDataNode and try to read from the bucket.
3. Check if data read by S3ObjectDataNode is the correct data stored in the bucket.

If not, the test should just be renamed as `test_read_write`.

Reply:
@trgiangdo, this seems fair, the 3 step approach makes more sense, but the bucket will be declared as a Gobal variable , is this okay with you ?

Reply:
why is it need to be a global variable?

Reply:
Your logic is right and straight forward. I don't know why I keep thinking of globals. Don't mid me :)